### PR TITLE
Notification queue

### PR DESCRIPTION
**Pull request by atpatil11**
Task #102428 - TJ Notification - CLI script to send Notifications (Dqueue notifications)

### CODE REVIEW

File: src/cli/notificationsProcessQueue.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	 * @since   1.0.0
+	 */
+	public function execute()
+	{
+		$db			=	JFactory::getDBO();
+		$mainframe	=	JFactory::getApplication('site');
+		$query		=	$db->getQuery(true);
+		$limit		=	$this->input->get('limit', 20, 'INT');
+
+		$query->select('*');
```
Comment (coolbung) [inline src/cli/notificationsProcessQueue.php]: There needs to be a queue model, no direct queries

File: src/cli/notificationsProcessQueue.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		$from		=	$mainframe->getCfg('mailfrom');
+		$fromname	=	$mainframe->getCfg('fromname');
+		$sentCount	=	0;
+
+		foreach ($queueData as $data)
+		{
+			$options	= json_decode($data->options, true);
+			$mailer		= JFactory::getMailer();
+
+			if ($options['emailfrom'] != null && $options['emailfromname'] != null)
```
Comment (coolbung) [inline src/cli/notificationsProcessQueue.php]: You're assuming that only emails will go in queue. What about sms, push and other types ?

Comment (atpatil11) [inline src/cli/notificationsProcessQueue.php]: @coolbung I'll add switch case to handle other notification types.

File: src/cli/notificationsProcessQueue.php (hunk 1 above)
Comment (atpatil11) [inline src/cli/notificationsProcessQueue.php]: @coolbung ok. Will use queue model.

**Comment by coolbung** [comment]
This also needs to have the SQL statements for the new table.

File: src/cli/notificationsProcessQueue.php (hunk 2 above)
Comment (coolbung) [inline src/cli/notificationsProcessQueue.php]: No that wont work. You have to use the adapter.

File: src/cli/notificationsProcessQueue.php (hunk 1 above)
Comment (vijaykhollam) [reply src/cli/notificationsProcessQueue.php]: is any one working on this?

**Pull request closed without merging**